### ISSUE TITLE
the hyperparameters are inconsistent with the paper

### ISSUE THREAD

**Issue by Ideal-Bushgorman** [ShashankHuang/RecLMIS#6]
Hi! Thank you for the awesome work. I carefully reviewed the code and noticed the hyperparameters in Config_MosMedPlus.py are  inconsistent with those reported in the paper. I also failed can't reproduce the experimental result. Here is my results on MosMedplus dataset:

![Image](https://github.com/user-attachments/assets/9cca8ef2-426d-462b-b19d-b24c4d02837d)

**Comment by ShawnHuang497** [comment]
Thank you for your attention to our work.

Answer to Q1: The reconstruction strategy we proposed includes several hyperparameters, such as mask ratio, the number of reconstruction layers, and so on. The optimal values of these hyperparameters may vary depending on the type and distribution of the dataset. In our paper, we primarily reported the parameters for the Covid19 dataset. For the MosMedplus dataset, which contains a large number of small targets, the optimal values of the hyperparameters have changed accordingly.

Answer to Q2: We also found during our experiments that the MosMedplus dataset is quite sensitive to the version of the GPU card, the version of the experimental virtual environment, and all hyperparameters. Unfortunately, due to a server failure and subsequent system reinstallation (which cleared the C drive), we lost the specific environment versions used at that time. However, the trained model in D drive has been open-sourced. You can provide an email address, and I can send you the training logs if you need. You can try reproducing the results using a 3090 GPU and the LViT environment.

Please feel free to let me know if you have more questions.

**Comment by Ideal-Bushgorman** [comment]
> Thank you for your attention to our work.
> 
> Answer to Q1: The reconstruction strategy we proposed includes several hyperparameters, such as mask ratio, the number of reconstruction layers, and so on. The optimal values of these hyperparameters may vary depending on the type and distribution of the dataset. In our paper, we primarily reported the parameters for the Covid19 dataset. For the MosMedplus dataset, which contains a large number of small targets, the optimal values of the hyperparameters have changed accordingly.
> 
> Answer to Q2: We also found during our experiments that the MosMedplus dataset is quite sensitive to the version of the GPU card, the version of the experimental virtual environment, and all hyperparameters. Unfortunately, due to a server failure and subsequent system reinstallation (which cleared the C drive), we lost the specific environment versions used at that time. However, the trained model in D drive has been open-sourced. You can provide an email address, and I can send you the training logs if you need. You can try reproducing the results using a 3090 GPU and the LViT environment.
> 
> Please feel free to let me know if you have more questions.

Thank you for your prompt reply!

**Comment by wh0am1s** [comment]
> Thank you for your attention to our work.
> 
> Answer to Q1: The reconstruction strategy we proposed includes several hyperparameters, such as mask ratio, the number of reconstruction layers, and so on. The optimal values of these hyperparameters may vary depending on the type and distribution of the dataset. In our paper, we primarily reported the parameters for the Covid19 dataset. For the MosMedplus dataset, which contains a large number of small targets, the optimal values of the hyperparameters have changed accordingly.
> 
> Answer to Q2: We also found during our experiments that the MosMedplus dataset is quite sensitive to the version of the GPU card, the version of the experimental virtual environment, and all hyperparameters. Unfortunately, due to a server failure and subsequent system reinstallation (which cleared the C drive), we lost the specific environment versions used at that time. However, the trained model in D drive has been open-sourced. You can provide an email address, and I can send you the training logs if you need. You can try reproducing the results using a 3090 GPU and the LViT environment.
> 
> Please feel free to let me know if you have more questions.

![Image](https://github.com/user-attachments/assets/526b6e09-5e70-4a81-bd69-89b82c1952bf)

> Thank you for your attention to our work.
> 
> Answer to Q1: The reconstruction strategy we proposed includes several hyperparameters, such as mask ratio, the number of reconstruction layers, and so on. The optimal values of these hyperparameters may vary depending on the type and distribution of the dataset. In our paper, we primarily reported the parameters for the Covid19 dataset. For the MosMedplus dataset, which contains a large number of small targets, the optimal values of the hyperparameters have changed accordingly.
> 
> Answer to Q2: We also found during our experiments that the MosMedplus dataset is quite sensitive to the version of the GPU card, the version of the experimental virtual environment, and all hyperparameters. Unfortunately, due to a server failure and subsequent system reinstallation (which cleared the C drive), we lost the specific environment versions used at that time. However, the trained model in D drive has been open-sourced. You can provide an email address, and I can send you the training logs if you need. You can try reproducing the results using a 3090 GPU and the LViT environment.
> 
> Please feel free to let me know if you have more questions.

Thank you for your excellent work. During my training process on the MosMedplus dataset, the best Mean Dice I achieved was 0.7166, and the test results were only around 75. I failed to reproduce the experimental results. I hope to receive your training logs to help me make adjustments. Thank you. Here is my email: shawz1010@outlook.com.

**Comment by wh0am1s** [comment]
> Thank you for your attention to our work.
> 
> Answer to Q1: The reconstruction strategy we proposed includes several hyperparameters, such as mask ratio, the number of reconstruction layers, and so on. The optimal values of these hyperparameters may vary depending on the type and distribution of the dataset. In our paper, we primarily reported the parameters for the Covid19 dataset. For the MosMedplus dataset, which contains a large number of small targets, the optimal values of the hyperparameters have changed accordingly.
> 
> Answer to Q2: We also found during our experiments that the MosMedplus dataset is quite sensitive to the version of the GPU card, the version of the experimental virtual environment, and all hyperparameters. Unfortunately, due to a server failure and subsequent system reinstallation (which cleared the C drive), we lost the specific environment versions used at that time. However, the trained model in D drive has been open-sourced. You can provide an email address, and I can send you the training logs if you need. You can try reproducing the results using a 3090 GPU and the LViT environment.
> 
> Please feel free to let me know if you have more questions.

![Image](https://github.com/user-attachments/assets/233a009c-b3cc-4b62-9535-7191d4dcd164)
